src/controllers/trigger.ts

```diff
@@ -1,105 +1,81 @@
+          res.status(400).json({
+            error: `category should be a string separated by , and can accept only ${Object.values(
+              PSICategories
+            ).join(', ')}`,
+          });
+          return;
         }
-      }) as PSICategories[];
-      if (chosenCategory?.length === 0) {
+      } else {
         res.status(400).json({
           error: `category should be a string separated by , and can accept only ${Object.values(
             PSICategories
           ).join(', ')}`,
         });
         return;
       }
-    } else {
-      res.status(400).json({
-        error: `category should be a string separated by , and can accept only ${Object.values(
-          PSICategories
-        ).join(', ')}`,
-      });
-      return;
     }
-  }
-
-  let chosenStartegy: PSIStrategy = defaultStrategy;
-  if (strategy && Object.values(PSIStrategy).includes(strategy as PSIStrategy)) {
-    chosenStartegy = strategy as PSIStrategy;
-  }
 
-  const { urls, alertConfig } = config;
-
-  const queries = urls.map(url => setUpLighthouseQueryString(url, chosenCategory, chosenStartegy));
-  // trigger the queries
-  const data = await Promise.allSettled(
-    queries.map(async query => {
-      const response = await (await fetch(query)).json();
-      if (response.error) {
-        throw new Error(response.error.message);
-      }
-      const { lighthouseResult } = response;
-      return getScore(lighthouseResult);
-    })
-  );
-  const report = data.map((result, index) => {
-    const url = urls[index];
-    if (result.status === 'fulfilled') {
-      const score = result.value;
-      return { url, ...score };
-    } else {
-      return { url, error: result.reason.message, failed: true };
+    let chosenStartegy: PSIStrategy = defaultStrategy;
+    if (strategy && Object.values(PSIStrategy).includes(strategy as PSIStrategy)) {
+      chosenStartegy = strategy as PSIStrategy;
     }
-  });
+    const { urls, alertConfig } = config;
 
-  const onlyAlertIfBelowBaseline = true; // set this to true if you want to send alert only if the score is below the baseline
-  const baseline = getBaselineService(apiKey.toString());
-  const result = compareReportWithBaseline(report, baseline, chosenCategory, chosenStartegy);
-  const emailAlertStatus = await sendAlertMail(
-    alertConfig,
-    result,
-    chosenStartegy,
-    onlyAlertIfBelowBaseline
-  );
-  const slackAlertStatus = await sendAlertToSlackChannel(
-    alertConfig,
-    result,
-    chosenStartegy,
-    onlyAlertIfBelowBaseline
-  );
-  res.json({
-    result,
-    report,
-    alertStatus: {
-      email: emailAlertStatus,
-      slack: slackAlertStatus,
-    },
-  });
+    const clientId = apiKey.toString();
+    await createQueue(QUEUE_NAME);
+    const message = createMessage(
+      { urls, apiKey, clientId, alertConfig, chosenStartegy, chosenCategory },
+      QUEUE_NAME
+    );
+    const msgId = (await produceMessage(message)) as string;
```

Comment:
somehow associate the `msgId` and `apiKey`, so that user can poll for status of the processing without having a socket connection as well